OpenDreamRuntime/Procs/DebugAdapter/DreamDebugManager.cs

```diff
@@ -12,12 +14,57 @@ sealed class DreamDebugManager : IDreamDebugManager {
     [Dependency] private readonly IProcScheduler _procScheduler = default!;
     [Dependency] private readonly IBaseServer _server = default!;
 
+    // Setup
     private DebugAdapter? _adapter;
-    private readonly Dictionary<string, List<Breakpoint>> _breakpoints = new();
+    private string RootPath => _resourceManager.RootPath ?? throw new Exception("No RootPath yet!");
+    private bool _stopOnEntry = false;
+
+    // State
+    private bool Stopped = false;
+    private bool _terminated = false;
+    private DreamThread? stoppedThread;
+
+    private enum StepMode {
+        None,
+        StepOver,  // aka "next"
+        StepIn,
+        StepOut,
+    }
+    private readonly Dictionary<int, (StepMode Mode, int FrameId)> threadStepModes = new();
+
+    // Breakpoint storage
+    private const string ExceptionFilterRuntimes = "runtimes";
+    private bool breakOnRuntimes = true;
+
+    private class FileBreakpointSlot {
```

Comment:
All classes must be marked as `[Virtual]`, `abstract`, `static`, or `sealed`. This is imposed by a linter in RobustToolbox.
```suggestion
    private sealed class FileBreakpointSlot {
```